subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/measure/DataSeries.java

```diff
@@ -56,6 +62,11 @@ public DataSeries(Iterable<? extends Amount<Q>> values) {
             min = min.compareTo(amount) <= 0 ? min : amount;
             max = max.compareTo(amount) >= 0 ? max : amount;
         }
+        List<Amount<Q>> sorted = Lists.newArrayList(this);
+        Collections.sort(sorted);
+        Amount<Q> medianLeft = sorted.get((sorted.size() - 1) / 2);
+        Amount<Q> medianRight = sorted.get((sorted.size() - 1) / 2 + 1 - sorted.size() % 2);
+        median = medianLeft.plus(medianRight).div(2);
```

Comment:
If there are an even number of points the median is indeed the average of the two middle points, but if there's an odd number of points the median is just the middle point.